plugins/modules/foreman_role.py

```diff
@@ -120,19 +128,29 @@ def main():
 
     filters = entity_dict.pop("filters", None)
 
-    changed, entity = module.ensure_entity('roles', entity_dict, entity)
-
-    if not module.desired_absent and not module.check_mode:
-        if filters is not None:
-            for role_filter in filters:
-                role_filter['role_id'] = entity['id']
-                role_filter['permissions'] = module.find_resources_by_name('permissions', role_filter['permissions'], thin=True)
-                module.ensure_entity_state('filters', role_filter, None, None, 'present', filters_entity_spec)
-            old_filters = entity.get('filter_ids', [])
-            for old_filter in old_filters:
-                module.ensure_entity('filters', None, {'id': old_filter}, {}, 'absent')
-            if len(old_filters) != len(filters):
-                changed = True
+    changed, new_entity = module.ensure_entity('roles', entity_dict, entity)
+
+    if not module.desired_absent and filters is not None:
+        scope = {'role_id': new_entity['id']}
+
+        if entity:
+            current_filters = [module.show_resource('filters', filter['id']) for filter in entity['filters']]
+        else:
+            current_filters = []
+        desired_filters = [filter.copy() for filter in filters]
```

Comment:
`copy.deepcopy(filters)`?

Reply:
That's about the level of copying i need, but then again, there will probably be no more levels.